extensions/src/ACRE2Arma/arma2ts/CallExt_DllMain.cpp

```diff
@@ -188,31 +187,39 @@ void __stdcall RVExtension(char *output, int outputSize, const char *function)
         case PIPE_COMMAND_READ: {
             if (readConnected) {
                 DWORD cbRead;
-                DWORD err;
                 BOOL ret;
-                char *value = new char[4096];
+                char value[4097]; // Allocate on stack to delegate memory management to compiler,
+                // allows up to 4096 char string (+1 to ensure NUL terminated), like initial version
+
                 //DEBUG("Read from pipe [%d]\n", hPipe);
-                ret = ReadFile(readHandle, (LPVOID)value, 4096, &cbRead, NULL);
-                if ( ! ret) {
-                    err = GetLastError();
-                    if (err == ERROR_BROKEN_PIPE) {
-                        ClosePipe();
-                    }
+                ret = ReadFile(readHandle, (LPVOID)value, sizeof(value)-1, &cbRead, NULL);
```

Comment:
```suggestion
                ret = ReadFile(readHandle, (LPVOID)value, (sizeof(char) - 1U) * read_length, &cbRead, NULL);
```

Reply:
I think yout meant `sizeof(char) * (read_length - 1U)` (otherwise it would be equal to `0`).